interface.py

```diff
@@ -572,51 +574,53 @@ def upload():
 
     files = request.files.getlist("file[]")
     if not files:
-        return redirect("./", code=400)
+        abort(400)
 
-    # filename = secure_filename(file.filename).strip()
     for file in files:
         filename = file.filename
         if filename == '':
-            return redirect("./", code=400)
+            abort(400)
 
         targetdir = request.form['targetdir'].strip()
         if targetdir == '':
             targetdir = 'uploads/'
         elif '../' in targetdir:
-            return redirect("./", code=400)
+            abort(403)
```

Comment:
Hahaha I love this! Yes, indeed, it should be 403 Forbidden :D

Reply:
Actually, I think this check is better if done outside the `for file in files:` loop, because `targetdir` is the same for all of them, right? And therefore you can check just once and know if the directory is valid for all of them :)